### DISCUSSION TITLE
sensor battery remain energy

### DISCUSSION THREAD

**Discussion by matomatusov** [slipx06/sunsynk-power-flow-card#295] [SolaX]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Hi,
If I use the battery remaining energy sensor, the data is displayed incorrectly.

![sensor battery remain energy](https://github.com/slipx06/sunsynk-power-flow-card/assets/129114957/b1aef8e7-f592-4081-accd-14c9a854ca08)

### Expected behaviour

repair of remaining energy

### Card Version

4.16.2

### Home Assistant Version

2024.2.3

**Comment by slipx06** [comment]
Energy should be the total battery energy in Wh. You can provide a sensor or just enter the value i.e. 10000 etc The remaining energy will be calculated for you based on the battery SOC and total battery energy.

 https://slipx06.github.io/sunsynk-power-flow-card/configuration.html#battery

**Comment by Roving-Ronin** [comment]
@matomatusov 

Have you had a look at the example for how to configure the card for SolaX inverters / batteries, at: https://slipx06.github.io/sunsynk-power-flow-card/examples/solax.html and also compare to options for batteries at: https://slipx06.github.io/sunsynk-power-flow-card/configuration.html#battery

For Solax options are:
```
  battery:
    energy: 17600          <--- Size of battery in Watts or (if) there's a sensor that reports the battery capacity.
    shutdown_soc: 11
    show_daily: true
    invert_power: true
    max_power: 3750
    auto_scale: false
    show_absolute: false
    hide_soc: false
```

Looking at the name of the sensor your using for energy: sensor.solax_battery_remaining_energy are you sure this is the capacity of the battery (i.e. when 100%), or what "remaining" in it at that moment ? 

Might be worth confirming if your using an incorrect sensor, by entering the watts value of the battery capacity (copy and paste the code above to replace your battery settings (adjust the energy watts to suit your battery model/size)?

Assuming your getting your data using this integration ( https://homeassistant-solax-modbus.readthedocs.io/en/latest/solax-entity-description/ ) the next step would be to see if sensor.battery_capacity is the one that gives you the correct energy capacity (like the 17600 watts example).

Re:  full_capacity: sensor.solax_battery_charge_upper_soc  checking the Sunsynk documentation and there doesnt appear to be such an entity as 'full_capacity' for any battery in the Sunsynk card looking at the documentation.

Hard to follow as the integration you appear to be using to pull the modbus data is rather lacking in regards its documentation and what each sensor is supposed to do. Hope the above helps anyhow.. or at worst just use the above yaml (that you've updated for your specific battery size / minimum SoC / shutdown SoC % etc.
